File: java/src/main/java/com/workshop/architecture/fitness/CustomerEntity.java
package com.workshop.architecture.fitness;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.LocalDate;
import java.util.UUID;

@Entity
@Table(name = "customers")
public class CustomerEntity {

    @Id
    private UUID id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private LocalDate dateOfBirth;

    @Column(nullable = false, unique = true)
    private String emailAddress;

    protected CustomerEntity() {
    }

    public CustomerEntity(UUID id, String name, LocalDate dateOfBirth, String emailAddress) {
        this.id = id;
        this.name = name;
        this.dateOfBirth = dateOfBirth;
        this.emailAddress = emailAddress;
    }

    public UUID getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public LocalDate getDateOfBirth() {
        return dateOfBirth;
    }

    public String getEmailAddress() {
        return emailAddress;
    }

    public void updateFrom(CustomerUpsertRequest request) {
        this.name = request.name();
        this.dateOfBirth = request.dateOfBirth();
        this.emailAddress = request.emailAddress();
    }
}

File: python/src/workshop_api/fitness/layered/infrastructure/customer_entity.py
from __future__ import annotations

import uuid

from sqlalchemy import Column, Date, String

from workshop_api.fitness.layered.infrastructure.database import Base


class CustomerOrmModel(Base):
    __tablename__ = "customers"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    name = Column(String(255), nullable=False)
    date_of_birth = Column(Date, nullable=False)
    email_address = Column(String(255), nullable=False, unique=True)
